### PR TITLE
[MAINTENANCE] Use DataContext to ignore progress bars

### PR DESCRIPTION
**Pull request by kenwade4**
Changes proposed in this pull request:
- Update build_gallery.py to use a DataContext to ignore progress bars
- Remove force_no_progress_bar kwarg from many funcs

### CODE REVIEW

File: great_expectations/expectations/expectation.py [hunk 1]
```diff
@@ -951,11 +949,11 @@ def build_configuration(cls, *args, **kwargs):
     def run_diagnostics(
         self,
         raise_exceptions_for_backends: bool = False,
-        force_no_progress_bar: bool = True,
         ignore_suppress: bool = False,
         ignore_only_for: bool = False,
         debug_logger: Optional[logging.Logger] = None,
         only_consider_these_backends: Optional[List[str]] = None,
+        context: Optional[Any] = None,  # Cannot type DataContext due to circular import
```
Comment (alexsherstinsky) [inline great_expectations/expectations/expectation.py]: @kenwade4 You can type it in quotes and then also add `# noqa: F812` -- here is an example:
```
Optional["Validator"]:  # noqa: F821
```
Also, please feel free to use the proper exact `DataContext` class (is it really `DataContext`?  or is it `AbstractDataContext`, etc.?). 

Thanks!

File: great_expectations/self_check/util.py [hunk 2]
```diff
@@ -1205,6 +1208,7 @@ def get_test_validator_with_data(
     sqlite_db_path=None,
     extra_debug_info="",
     debug_logger: Optional[logging.Logger] = None,
+    context: Optional[Any] = None,
```
Comment (alexsherstinsky) [inline great_expectations/self_check/util.py]: @kenwade4 Same comment as above, regarding the context type.  Thanks!

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
@kenwade4 Minor comments -- thanks so much! /cc @cdkini

File: great_expectations/expectations/expectation.py (hunk 1 above)
Comment (kenwade4) [reply great_expectations/expectations/expectation.py]: Yea it's a full-on DataContext object returned from `DataContext.create(...)`

File: assets/scripts/build_gallery.py [hunk 3]
```diff
@@ -603,6 +624,9 @@ def main(**kwargs):
     with open(f"./{kwargs['outfile_name']}", "w") as outfile:
         json.dump(gallery_info, outfile, indent=4)
 
+    print(f"Deleting {context_dir}")
+    shutil.rmtree(context_dir)
+
```
Comment (cdkini) [inline assets/scripts/build_gallery.py:629]: should we maybe `try/except/finally` this so we always delete?

**Review by alexsherstinsky — APPROVED** [review]
LGTM (if @cdkini also approves).  Thanks!

**Pull request merged**